server.go

```diff
@@ -240,11 +250,24 @@ func (m *Serve) getService(sd []proxy.ServiceDest, req *http.Request) proxy.Serv
 	sr.SkipCheck = m.getBoolParam(req, "skipCheck")
 	sr.Distribute = m.getBoolParam(req, "distribute")
 	sr.SslVerifyNone = m.getBoolParam(req, "sslVerifyNone")
+	sr.UsersPassEncrypted = m.getBoolParam(req, "usersPassEncrypted")
+
 	if len(req.URL.Query().Get("users")) > 0 {
-		users := strings.Split(req.URL.Query().Get("users"), ",")
-		for _, user := range users {
-			userPass := strings.Split(user, ":")
-			sr.Users = append(sr.Users, proxy.User{Username: userPass[0], Password: userPass[1]})
+		appendUsersFromString(&sr, req.URL.Query().Get("users"))
+	} else if len(req.URL.Query().Get("usersPath")) > 0 {
```

Comment:
Should there be the option of storing users in a file since Docker Secrets already has that feature? Maybe change to `usersSecret` param with only a file name (dir would be `/run/secrets`). I know it's less flexible than to use any path so my proposal is, in a way, downgrade from what you did. I'm only wondering whether we should enforce some (hopefully good) patterns or go for flexibility. We can even have a predefined file as users secret so that there is no need to specify `usersSecret`. What do you think?

Another thing to consider is how will that file get into proxy containers? It needs to be there before the service that needs those users is created. If a volume is mounted, that users would need to add the file first. If Docker secret is used, it would need to be updated with additional users. A new secret would restart the proxy service so it would need to be an update of the existing secret. Maybe users can be in the request in both cases. If password is not present (e.g. `myuser` NOT `myuser:mypassword`) it could be read from a secret file. It would mean that the file should be converted into a map and matched with users from the query. That would make it more secure and would allow people to have a central location with their credentials (secrets file) but decide from one service to another which users should be used.

I just wrote things that came to my mind without much deep thinking so some of those comments might be partly (or completely) useless. The main reason was to initiate a discussion.